lib/Runtime/Library/TypedArray.cpp

```diff
@@ -606,15 +606,39 @@ namespace Js
             object;
     };
 
-    BOOL TypedArrayBase::HasProperty(PropertyId propertyId)
+    BOOL TypedArrayBase::HasOwnProperty(PropertyId propertyId)
+    {
+        return HasPropertyEx(propertyId) == Property_Found;
+    }
+
+    inline BOOL TypedArrayBase::CanonicalNumericIndexString(PropertyId propertyId, ScriptContext *scriptContext)
+    {
+        PropertyString *propertyString = scriptContext->GetPropertyString(propertyId);
```

Comment:
I took away the unnecessary type checks in JavascriptConversion::CanonicalNumericIndexString to only use the primitive operations.

---
In reply to: [105423771](https://github.com/Microsoft/ChakraCore/pull/2640#discussion_r105423771) [](ancestors = 105423771,105298966,105254349)